src/Compilers/Test/Core/Diagnostics/DiagnosticExtensions.cs

```diff
@@ -314,12 +314,15 @@ public static TCompilation VerifyDiagnostics<TCompilation>(this TCompilation c,
                 analyzerExceptionFilter: null, reportAnalyzer: false, severityFilter: SeverityFilter.None, trackSuppressedDiagnosticIds: false,
                 out var newCompilation, cancellationToken);
             Debug.Assert(newCompilation.SemanticModelProvider != null);
-            var compilerDiagnostics = newCompilation.GetDiagnostics(cancellationToken);
-            var analyzerDiagnostics = driver.GetDiagnosticsAsync(newCompilation, cancellationToken).Result;
-            var allDiagnostics = includeCompilerDiagnostics ?
-                compilerDiagnostics.AddRange(analyzerDiagnostics) :
-                analyzerDiagnostics;
-            diagnostics = driver.ApplyProgrammaticSuppressionsAndFilterDiagnostics(allDiagnostics, newCompilation, cancellationToken);
+            diagnostics = TestHelpers.WithCulture(() =>
```

Comment:
@AlekseyTs the use of `TestHelpers.WithCulture` in this file is meant to force `CurrentUICulture` to our loc culture when getting diagnostics. This will help us catch cases when we inadvertently realize resource strings directly into a diagnostic instead of using a localizable argument. This is what allowed me to see #71236 locally and in CI. 

Happy to hear any suggestions you have about better ways to plug into our diagnostic test infra.